runtime/runtime.go

```diff
@@ -147,11 +147,22 @@ func list() []string {
 	return ret
 }
 
-func register(failpath string, fp *Failpoint) {
+func register(name string) *Failpoint {
 	failpointsMu.Lock()
-	failpoints[failpath] = fp
-	failpointsMu.Unlock()
-	if t, ok := envTerms[failpath]; ok {
-		Enable(failpath, t)
+	defer failpointsMu.Unlock()
+	if f, ok := failpoints[name]; ok {
+		fmt.Fprintf(
+			os.Stderr,
+			"warning: failpoint name %s is already registered. You are using same failpoint name in multiple packages.\n",
+			name)
+		// return existing failpoint, don't re-enable
+		return f
```

Comment:
We shouldn't allow duplicated failpoint, so we might need to panic here? WDYT? cc @serathius

Reply:
Yep, we definitely need to panic here.